docs/notebooks/proton_radiography_particle_tracing_custom_source.ipynb

```diff
@@ -0,0 +1,524 @@
+    "# Create a spherical potential with a Gaussian radial distribution\n",
+    "radius = np.linalg.norm(grid.grid, axis=3)\n",
+    "arg = (radius / (L / 3)).to(u.dimensionless_unscaled)\n",
+    "potential = 6e5 * np.exp(-(arg ** 2)) * u.V\n",
+    "\n",
+    "# Calculate E from the potential\n",
+    "Ex, Ey, Ez = np.gradient(potential, grid.dax0, grid.dax1, grid.dax2)\n",
+    "Ex = -np.where(radius < L / 2, Ex, 0)\n",
+    "Ey = -np.where(radius < L / 2, Ey, 0)\n",
+    "Ez = -np.where(radius < L / 2, Ez, 0)\n",
+    "\n",
+    "# Add those quantities to the grid\n",
+    "grid.add_quantities(E_x=Ex, E_y=Ey, E_z=Ez, phi=potential)\n",
+    "\n",
+    "\n",
+    "# Plot the E-field\n",
+    "fig = plt.figure(figsize=(8, 8))\n",
+    "ax = fig.add_subplot(111, projection=\"3d\")\n",
+    "ax.view_init(30, 30)\n",
+    "\n",
+    "# skip some points to make the vector plot intelligable\n",
+    "s = tuple([slice(None, None, 6)] * 3)\n",
+    "\n",
+    "ax.quiver(\n",
+    "    grid.pts0[s].to(u.mm).value,\n",
+    "    grid.pts1[s].to(u.mm).value,\n",
+    "    grid.pts2[s].to(u.mm).value,\n",
+    "    grid[\"E_x\"][s],\n",
+    "    grid[\"E_y\"][s],\n",
+    "    grid[\"E_z\"][s],\n",
+    "    length=5e-7,\n",
+    ")\n",
+    "\n",
+    "ax.set_xlabel(\"X (mm)\")\n",
+    "ax.set_ylabel(\"Y (mm)\")\n",
+    "ax.set_zlabel(\"Z (mm)\")\n",
+    "ax.set_xlim(-1, 1)\n",
+    "ax.set_ylim(-1, 1)\n",
+    "ax.set_zlim(-1, 1)\n",
+    "ax.set_title(\"Gaussian Potential Electric Field\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will then create the synthetic radiograph object, specifying the remaining setup geometry.\n",
+    "\n",
+    "The warning filter ignores a warning that arises because $B_x$, $B_y$, $B_z$ are not provided in the grid (they will be assumed to be zero)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "detector = (0 * u.mm, 0 * u.mm, 100 * u.mm)\n",
+    "\n",
+    "with warnings.catch_warnings():\n",
+    "    warnings.simplefilter(\"ignore\")\n",
+    "    sim = prad.SyntheticProtonRadiograph(grid, source, detector, verbose=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, instead of using `create_particles` to create the particle distribution, we will use the `load_particles` function to use the particles we have created above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sim.load_particles(pos, vel, particle=particle)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After this point, the rest of the process for creating the radiograph is unchanged."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "nbsphinx-thumbnail": {
+     "tooltip": "Proton Radiography"
+    }
```

Comment:
For some reason this thumbnail tagging is not working.  I'll continue looking into it, but two pairs will be quicker than one.

Reply:
Hm, no idea, I checked and this `metadata` is identical to what I have in the main proton radiography notebook that seems to be working.

Reply:
I did the same and couldn't see why.  I'll keep investigating.  I'm sure we'll find the culprit as we get through the review.

Reply:
Ha, third pair's the charm! At first I thought this was a question of [multiple outputs from a cell](https://nbsphinx.readthedocs.io/en/0.8.2/gallery/multiple-outputs.html), and it kind of is. tqdm is the issue - "stream" output seems to be the on-the-fly progress bar animation. Every text line of the ASCII progress bar seems to be recorded as an output.

To sidestep this, I just split this cell into one for the simulation and one for the plot.